File: patch_cadence.js
const fs = require('fs');
const file = 'src/game/systems/platformOriginalsReleaseCadenceSystem.ts';
let content = fs.readFileSync(file, 'utf8');

content = content.replace(
  /const premiereDate = active\.season\.premiereDate;/g,
  `const premiereDate = active.season.premiereDate ?? fromAbs(active.premiereAbs);`
);

content = content.replace(
  /const finaleDate = active\.season\.finaleDate;/g,
  `const finaleDate = nextAired >= active.total ? (active.season.finaleDate ?? fromAbs(finaleAbs)) : active.season.finaleDate;`
);

fs.writeFileSync(file, content);

File: patch_cadence.py
import re

with open('src/game/systems/platformOriginalsReleaseCadenceSystem.ts', 'r') as f:
    content = f.read()

pattern = r'''      expectedAired = clampInt\(expectedAired, 0, active\.total\);

      const nextAired = active\.season\.episodesAired \?\? 0;'''

replacement = '''      expectedAired = clampInt(expectedAired, 0, active.total);

      const prevAired = clampInt(active.season.episodesAired ?? 0, 0, active.total);
      const nextAired = Math.max(prevAired, expectedAired);'''

content = re.sub(pattern, replacement, content)

with open('src/game/systems/platformOriginalsReleaseCadenceSystem.ts', 'w') as f:
    f.write(content)
